common/src/main/java/com/gregtechceu/gtceu/integration/kjs/builders/machine/CustomMultiblockBuilder.java

```diff
@@ -63,10 +101,254 @@ public static MachineBuilder<MultiblockMachineDefinition> createMultiblock(Strin
+                    builder.itemColor(itemColor);
+                }
+                return this;
+            }
+
+            @Override
+            public CustomMultiblockBuilder abilities(PartAbility... abilities) {
+                for (var builder : builders) {
+                    builder.abilities(abilities);
+                }
+                return this;
+            }
+
+            public CustomMultiblockBuilder tooltips(Component... tooltips) {
+                for (var builder : builders) {
+                    builder.tooltips(tooltips);
+                }
+                return this;
+            }
+
+            @Override
+            public CustomMultiblockBuilder tooltipBuilder(BiConsumer<ItemStack, List<Component>> tooltipBuilder) {
+                for (var builder : builders) {
+                    builder.tooltipBuilder(tooltipBuilder);
+                }
+                return this;
+            }
+
+            @Override
+            public CustomMultiblockBuilder recipeModifier(BiFunction<MetaMachine, GTRecipe, GTRecipe> recipeModifier) {
+                for (var builder : builders) {
+                    builder.recipeModifier(recipeModifier);
+                }
+                return this;
+            }
+
+            @Override
+            public CustomMultiblockBuilder alwaysTryModifyRecipe(boolean alwaysTryModifyRecipe) {
+                for (var builder : builders) {
+                    builder.alwaysTryModifyRecipe(alwaysTryModifyRecipe);
+                }
+                return this;
+            }
+
+            @Override
+            public CustomMultiblockBuilder appearance(Supplier<BlockState> appearance) {
+                for (var builder : builders) {
+                    builder.appearance(appearance);
+                }
+                return this;
+            }
+
+            @Override
+            public CustomMultiblockBuilder editableUI(@Nullable EditableMachineUI editableUI) {
+                for (var builder : builders) {
+                    builder.editableUI(editableUI);
+                }
+                return this;
+            }
+
+            @Override
+            public CustomMultiblockBuilder langValue(String langValue) {
+                for (var builder : builders) {
+                    builder.langValue(langValue);
+                }
+                return this;
+            }
+
+            public CustomMultiblockBuilder recipeModifier(BiFunction<MetaMachine, GTRecipe, GTRecipe> recipeModifier, boolean alwaysTryModifyRecipe) {
+                recipeModifier(recipeModifier);
+                alwaysTryModifyRecipe(alwaysTryModifyRecipe);
+                return this;
+            }
+
+            @Override
+            public MultiblockMachineDefinition register() {
+                for (var builder : builders) {
+                    value = builder.register();
+                }
+                return value;
+            }
+        };
+    }
```

Comment:
(tier, MultiblockMachineDefinition) => {
}

Reply:
Avoid repeating code.

Reply:
Reduce the work needed to extend builder functionality each time. Not everyone remembers that there is another kjs class that needs to be copied exactly the same methods

Reply:
That isn't the case. I just couldn't do it with the generic builder, because that doesn't have the multiblock-specific parts.

Reply:
How would i fix that, anyway?

Reply:
> (tier, MultiblockMachineDefinition) => {
> }

Also: can't do this without just giving out a consumer, which i don't exactly like the idea of.

Reply:
I like it

Reply:
@FunctionInterance